packages/next-swc/crates/next-core/js/src/overlay/internal/container/Errors.tsx

```diff
@@ -130,221 +130,287 @@ const enum DisplayState {
-    onlyHasWarnings ? DisplayState.Minimized : DisplayState.Fullscreen
-  );
-  let displayState = stateDisplayState;
+  // Toast is hidden
+  const [hidden, setHidden] = React.useState(false)
 
-  if (!isClosable) {
-    displayState = DisplayState.Fullscreen;
-  }
+  // Already seen issue ids, to auto open the dialog on new errors
+  const [seenIds, setSeenIds] = React.useState(() => new Set())
+
+  React.useEffect(() => {
+    const newSeenIds = new Set()
+    let change = false
+    let autoOpen = false
+    // When the selected tab disappears we will go to another important tab or close the overlay
+    if (selectedTab && !tabs.some((tab) => tab.id === selectedTab)) {
+      const otherImportantTab = tabs.find((tab) => tab.autoOpen)
+      if (otherImportantTab) {
+        setSelectedTab(otherImportantTab.id)
+      } else {
+        setSelectedTab(null)
+      }
+    } else {
+      autoOpen = true
+    }
+    // When there is a new item we open the overlay when autoOpen is set
+    for (const tab of tabs) {
+      for (const item of tab.items) {
+        newSeenIds.add(itemHash(item))
+        if (!seenIds.has(itemHash(item))) {
+          change = true
+          setHidden(false)
+          if (autoOpen && tab.autoOpen) {
+            setSelectedTab(tab.id)
+          }
+        }
+      }
+    }
+    if (change || newSeenIds.size !== seenIds.size) setSeenIds(newSeenIds)
+  }, [selectedTab, tabs, seenIds])
 
   // This component shouldn't be rendered with no errors, but if it is, let's
   // handle it gracefully by rendering nothing.
-  if (!hasErrors && !hasTurbopackWarnings && !hasTurbopackErrors) {
-    return null;
+  if (tabs.length === 0) {
+    return null
   }
 
-  if (displayState === DisplayState.Hidden) {
-    return null;
+  if (hidden) {
+    return null
   }
 
-  if (displayState === DisplayState.Minimized) {
+  if (selectedTab === null || !tabs.some((tab) => tab.id === selectedTab)) {
+    const errors = tabs.reduce(
+      (sum, tab) => sum + (tab.severity === 'error' ? tab.items.length : 0),
+      0
+    )
+    const warnings = tabs.reduce(
+      (sum, tab) => sum + (tab.severity === 'warning' ? tab.items.length : 0),
+      0
+    )
+
+    if (errors === 0 && warnings === 0) return null
+
     return (
       <ErrorsToast
-        errorCount={readyErrors.length + turbopackErrors.length}
-        warningCount={turbopackWarnings.length}
-        severity={onlyHasWarnings ? "warning" : "error"}
-        onClick={fullscreen}
-        onClose={hide}
+        errorCount={errors}
+        warningCount={warnings}
+        severity={errors > 0 ? 'error' : 'warning'}
+        onClick={() => setSelectedTab(tabs[0].id)}
+        onClose={() => setHidden(true)}
       />
-    );
+    )
   }
 
   return (
     <ErrorsDialog
       aria-labelledby="nextjs__container_errors_label"
       aria-describedby="nextjs__container_errors_desc"
-      onClose={isClosable ? minimize : undefined}
+      onClose={() => setSelectedTab(null)}
```

Comment:
Not sure if I like this, this means if you minimize and reopen the overlay you'll be back to the first tab?